src/sidebar/util/test/copy-to-clipboard-test.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const { copyText } = require('../copy-to-clipboard');
+
+describe('copy-to-clipboard', () => {
+  beforeEach(() => {
+    sinon.stub(document, 'execCommand');
+  });
+
+  afterEach(() => {
+    document.execCommand.restore();
+  });
+
+  describe('copyText', () => {
+    /**
+     * Returns the temporary element used to hold text being copied.
+     */
+    function tempSpan() {
+      return document.querySelector('.copy-text');
+    }
+
+    it('copies the passed text to the clipboard', () => {
+      // We can't actually copy to the clipboard due to security restrictions,
+      // but we can verify that `execCommand("copy")` was called and that the
+      // passed text was selected at the time.
+      const testStr = `test string ${Math.random()}`;
+      document.execCommand.callsFake(() => {
+        assert.equal(document.getSelection().toString(), testStr);
+      });
+      copyText(testStr);
+      assert.calledWith(document.execCommand, 'copy');
+      assert.isNull(tempSpan());
+    });
+
+    it('removes temporary span if copying fails', () => {
+      document.execCommand.callsFake(() => {
+        assert.ok(tempSpan());
```

Comment:
This method of effectively injecting assertions into the scope of the tested function is interesting...but is it possibly a problematic pattern?

Reply:
The purpose of putting the asserts here is to check the state at the point when `execCommand` is called, rather than after the `copyText` function returns, at which point these asserts would fail.